benchmark/infrastructure/util/ANTLRLexerHelper.h

```diff
@@ -0,0 +1,66 @@
+// Copyright 2023, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Andre Schlegel (March of 2023, schlegea@informatik.uni-freiburg.de)
+
+#pragma once
+
+#include <ANTLRInputStream.h>
+#include <Lexer.h>
+
+#include <concepts>
+#include <string_view>
+
+#include "util/Algorithm.h"
+#include "util/json.h"
+
+/*
+@brief Check, if the given string, interpreted with the given lexer, only
+contains tokens with same token indexes and in the same order, as the given
+indexes.
+
+@tparam lexer A lexer created by antlr4. Warning: It WILL be reset.
+
+@param str A string, which content will be transformed into tokens with the
+given lexer.
+@param tokenTypeIds In `anltr4` every token type has it's own unique index for
+identfication. You should find yours as an enum inside your lexer class.
+IMPORTANT: Remember, the highest possible lexer rule will always be used by the
+lexer to generate a token. That INCLUDES rules, that are just `or` collections
+of other lexer rules. For example:
+```
+LITERAL : BOOL | INTEGER;
+BOOL : 'true' | 'false';
+INTEGER : '-'?[0-9]+;
+```
+With this grammar, you will NEVER get a token of type `BOOL`, or `INTEGER`,
+because `LITERAL` has higher priority and includes all the cases of the other
```

Comment:
What is then the input here? I need a more explizcit example to understand this comment.

Reply:
The input is not really relevant, but I've added an example input: [a672f50](https://github.com/ad-freiburg/qlever/pull/977/commits/a672f5097d0a4b15575793d740f993434c93a01a)